internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -705,12 +705,16 @@ func getPlatform(aciSpec hiveext.AgentClusterInstallSpec) (*models.Platform, err
 			Type: common.PlatformTypePtr(models.PlatformTypeBaremetal),
 		}, nil
 	case hiveext.ExternalPlatformType:
-		if aciSpec.ExternalPlatformSpec != nil && aciSpec.ExternalPlatformSpec.PlatformName == string(models.PlatformTypeOci) {
+		if aciSpec.ExternalPlatformSpec != nil && aciSpec.ExternalPlatformSpec.PlatformName == common.ExternalPlatformNameOci {
 			return &models.Platform{
-				Type: common.PlatformTypePtr(models.PlatformTypeOci),
+				Type: common.PlatformTypePtr(models.PlatformTypeExternal),
+				External: &models.PlatformExternal{
+					PlatformName:           &aciSpec.ExternalPlatformSpec.PlatformName,
+					CloudControllerManager: swag.String(models.PlatformExternalCloudControllerManagerExternal),
+				},
 			}, nil
 		} else {
-			return nil, errors.New(fmt.Sprintf("For external platform, the platformName must be %s", string(models.PlatformTypeOci)))
+			return nil, errors.New(fmt.Sprintf("For external platform, the platformName must be %s", common.ExternalPlatformNameOci))
```

Comment:
@pawanpinjarkar can you please review this change as I'm removing Oci platform type